### ISSUE TITLE
[FEATURE] Support default Databricks notebook authentication

### ISSUE THREAD

**Issue by yb-yu** [databricks/databricks-sdk-java#230]
**Problem Statement**

Hello, it automatically reads the generated personal access token on the databricks notebooks for authentication without requiring separate authentication when using the Python SDK.

Simply declaring a WorkspaceClient is enough to correctly receive the authentication of the entity running the notebook, whether it's run by a user or a service principal through a workflow. 

```
from databricks.sdk import WorkspaceClient

w = WorkspaceClient()

for c in w.clusters.list():
  print(c.cluster_name)
```

I wish a similar feature was available in Scala, but it seems like the Databricks Java SDK does not support this. 

Is there any plan to support this?

**Additional Context**

https://docs.databricks.com/en/dev-tools/sdk-python.html#use-the-databricks-sdk-for-python-from-a-databricks-notebook

**Comment by mgyucht** [comment]
This is implemented here: https://github.com/databricks/databricks-sdk-java/blob/main/databricks-sdk-java/src/main/java/com/databricks/sdk/core/NotebookNativeCredentialsProvider.java. Have you tried it, or is it not working for you?

**Comment by yb-yu** [comment]
Oh.. I just tried declaring empty config and using it, but it didn't work well, so I didn't look deeper into the code. 

But after seeing the code you attached and actually testing it, I found that setting the setHost directly in the notebook allows calls to be made without any separate credential settings. 

However, while there's no need to set the host when executing in the notebook with the Python SDK, it seems that's not the case with Java SDK.

It seems like it should work when looking at [here](https://github.com/databricks/databricks-sdk-java/blob/7820506311047ba1b13533aac3df523a0be1bc23/databricks-sdk-java/src/main/java/com/databricks/sdk/core/NotebookNativeCredentialsProvider.java#L113-L115)

```
# 
import com.databricks.sdk.service.compute.ListClustersRequest
import com.databricks.sdk.WorkspaceClient
import com.databricks.sdk.core.DatabricksConfig

// It works in databricks notebook!
val config = new DatabricksConfig()
    .setHost("https://krafton-hq.cloud.databricks.com")
val w = new WorkspaceClient(config)
w.clusters().list(new ListClustersRequest()).forEach{
  elem => println(elem.getClusterName)
}

// It doesn't works..
// Caused by: ClientProtocolException: 
// Caused by: ProtocolException: Target host is not specified
val config2 = new DatabricksConfig()
val w2 = new WorkspaceClient(config2)
w2.clusters().list(new ListClustersRequest()).forEach{
  elem => println(elem.getClusterName)
}
```

**Comment by mgyucht** [comment]
Awesome, thank you for this report & investigation! You shouldn't need to set the host, so this is definitely an issue in the SDK. We'll take it from here to investigate and make it work.

**Comment by yb-yu** [comment]
Thank you!

**Comment by pietern** [comment]
I found out why this didn't work and made a fix in #282.

TL;DR: the "host" field was read by the HTTP client before the credential provider had a chance to populate it.
